src/app/core/configurable-enum/configurable-enum-datatype/configurable-enum-datatype.ts

```diff
@@ -37,8 +37,9 @@ export class ConfigurableEnumDatatype
       enumId = CONFIGURABLE_ENUM_CONFIG_PREFIX + enumId;
     }
 
+    // The label check is used to also match enums that were automatically created
     return this.configService
       .getConfig<ConfigurableEnumConfig>(enumId)
-      ?.find((option) => option.id === value);
+      ?.find((option) => option.id === value || option.label === value);
```

Comment:
the label shouldn't match here, this makes the concept confusing and opens ways for bugs if stuff is renamed in a circular way over the years. if enums are auto-created, why not create the "correct" ID?

Reply:
The enums are auto created based on the configuration of how they are editable (e.g. through the config of entity subrecords). To fully migrate the instances, all the entities where the values are used have to be edited. The adaption here can be seen as a temporary workaround. Running the migration [here](https://github.com/Aam-Digital/ndb-core/pull/866/files#diff-96ec83c71bd1fe9599268c3a2bfc1352a59fae23c828e572497ecadf52cea94aR15) will automatically migrate all entities to the new format and afterwards this second option can be removed.